web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\usergreeting;
+
+use Drupal\Component\Datetime\TimeInterface;
+
+/**
+ * Provides a greeting service for users dependent on the time of  day.
+ */
+class GreetingTime {
+
+  /**
+   * The user account.
+   *
+   * @var \Drupal\usergreeting\Drupal\Component\Datetime\TimeInterface
+   */
+  protected $timeOutput;
+
+  /**
+   * Construct the Timeinterface object.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $timeOutput
+   *   The timestamp output.
+   */
+  public function __construct(TimeInterface $timeOutput) {
+    $this->timeOutput = $timeOutput;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function greetingMessage() {
+    $time_output = intval(date('H', $this->timeOutput->getRequestTime()));
```

Comment:
uset type casting instead.
```suggestion
    $time_output = (int) date('H', $this->timeOutput->getRequestTime());
```

Reply:
And don't forget to factor in the user's timezone when calculating the greeting.